### ISSUE TITLE
Add Pytorch's nn.Module LOADABLE_CLASSES

### ISSUE THREAD

**Issue by luisaforozco** [huggingface/diffusers#6132]
### Model/Pipeline/Scheduler description

In class [diffusers.DiffusionPipeline](https://github.com/huggingface/diffusers/blob/v0.24.0/src/diffusers/pipelines/pipeline_utils.py#L528) there is a method [`save_pretrained`](https://github.com/huggingface/diffusers/blob/76c645d3a641c879384afcb43496f0b7db8cc5cb/src/diffusers/pipelines/pipeline_utils.py#L609) that looks into the constant [`LOADABLE_CLASSES`](https://github.com/huggingface/diffusers/blob/76c645d3a641c879384afcb43496f0b7db8cc5cb/src/diffusers/pipelines/pipeline_utils.py#L85) for the methods to save and load a pretrained model.
I am wondering if we could add pytorch's methods to save a `torch.nn.Module`: [`torch.save` and `torch.load`](https://pytorch.org/tutorials/beginner/saving_loading_models.html#save-load-entire-model).

I am not sure if this issue belongs to this category. If this sound like a good idea, I can help with the implementation.

### Open source status

- [ ] The model implementation is available.
- [ ] The model weights are available (Only relevant if addition is not a scheduler).

**Comment by sayakpaul** [comment]
@patrickvonplaten @DN6 WDYT?

**Comment by DN6** [comment]
Hi @luisaforozco could you provide an example for how you were envisioning using this feature?

**Comment by luisaforozco** [comment]
Hello @DN6, I have copied the [train loop from the documentation](https://huggingface.co/docs/diffusers/tutorials/basic_training#train-the-model) and all the way at the end of `def train_loop` 
you have the line `pipeline.save_pretrained(config.output_dir)`.
I was running some experiments with some GNNs for molecules defined as `class GraphGNNModel(torch.nn.Module):` thus when getting to this line the model does not save and I get printed out the following:
```python
self.unet=GraphGNNModel(
  (GNN): GNNModel(
    (layers): ModuleList(
      (0): GCNConv(11, 128)
      (1): ReLU(inplace=True)
      (2): Dropout(p=0.1, inplace=False)
      (3): GCNConv(128, 128)
      (4): ReLU(inplace=True)
      (5): Dropout(p=0.1, inplace=False)
      (6): GCNConv(128, 4)
    )
  )
  (lattice_MLP): Sequential(
    (0): Dropout(p=0.5, inplace=False)
    (1): Linear(in_features=4, out_features=6, bias=True)
  )
) of type <class '__main__.GraphGNNModel'> cannot be saved.
```
I envision using this feature to use diffusers for the diffusion part, but be able to hook my own model that might not be in the hugging face models or has a different architecture than that of UNets.

**Comment by DN6** [comment]
@luisaforozco `save_pretrained` also saves a config.json for each model type so that it is easy to load with the pipelines. We do make a final call to `torch.save` to save the state dict. 

Would you consider wrapping your model in a class with a ModelMixin? Happy to revisit if there is more demand for this feature.

**Comment by DN6** [comment]
Can you also add `ConfigMixin` to your model class?

**Comment by luisaforozco** [comment]
Thanks, that works for saving models.

**Comment by sayakpaul** [comment]
Is this still a problem?
